gitindex/index.go

```diff
@@ -887,18 +887,19 @@ func createDocument(key fileKey,
 	opts build.Options,
 ) (zoekt.Document, error) {
 	blob, err := repos[key].Repo.BlobObject(key.ID)
+
+	// We filter out large documents when fetching the repo. So if an object is too large, it will not be found.
+	if errors.Is(err, plumbing.ErrObjectNotFound) {
```

Comment:
Because it's just the blobs that are excluded, we still traverse all files in the commit tree and report "skipped docs" correctly. I tested this locally by indexing `golang/go` with and without this change, and checking that the number of docs matching "NOT-INDEXED: file size exceeds maximum" was identical.